### ISSUE TITLE
evaluating 'donutItemListeners[i].removeAllListeners'

### ISSUE THREAD

**Issue by mahiruslu** [Novsochetra/react-native-circular-chart#1]
Hello,
I'm getting this error when a screen that has this component closes.
" TypeError: undefined is not an object (evaluating 'donutItemListeners[i].removeAllListeners') "
I mixed up but haven't solved the problem.

<img width="920" alt="image" src="https://user-images.githubusercontent.com/23373520/155593908-7f0d87b8-8aec-4fe9-90e1-339e70cd1d21.png">

**Comment by Novsochetra** [comment]
@mahiruslu can you show me your code ?

**Comment by mahiruslu** [comment]
As your code example actually.

        <DonutChart
          data={results}
          strokeWidth={25}
          radius={90}
          containerWidth={SIZES.width - 20}
          containerHeight={105 * 2}
          type="round"
          startAngle={210}
          endAngle={510}
          animationType="slide"
        />

**Comment by Novsochetra** [comment]
@mahiruslu I think the **results** variable is wrong format, please help provide me the **results** variable

**Comment by mahiruslu** [comment]
Okay, but the result variable same as your DATA format. 

  const results = [
    {name: 'Doğru', value: correctAnswers, color: '#00ff00'},
    {name: 'Yanlış', value: wrongAnswers, color: '#ff0000'},
    {name: 'Boş', value: totalQuestions - correctAnswers - wrongAnswers, color: '#ffff00'},
  ];

**Comment by jathnamas** [comment]
Hello, running into the same error currently, was this ever addressed?

**Comment by mahiruslu** [comment]
Hi, I can't remember exactly, but I guess I deleted "donutItemListeners[i].removeAllListeners" under the node_modules from package

**Comment by andreysmattos** [comment]
Hello, i'm struggling with a similar error:
```
...
ERROR  Warning: Internal React error: Attempted to capture a commit phase error inside a detached tree. This indicates a bug in React. Likely causes include deleting the same fiber more than once, committing an already-finished tree, or an inconsistent return pointer.

Error message:

TypeError: Cannot read property 'removeAllListeners' of undefined
    in DonutChart (created by DonutChartCustom)
    in DonutChartCustom (created by FitnessScreen)
    in RCTView (created by View)
```

I use this lib in another project with a older react-native version, using same data and it works well.

sorry my bad english

**Comment by Novsochetra** [comment]
hello @andreysmattos thank for using this lib. 

would you mind giving me the verison of react-native that you are using?

**Comment by Novsochetra** [comment]
I am closed this issues now, this issues should be fixed from version **1.0.9** onward. let me know if it still has the problems

**Comment by Novsochetra** [comment]
I am closed this issues now, this issues should be fixed from version 1.0.9 onward. let me know if it still has the problems.
